### ISSUE TITLE
MeshAgent Data Base File

### ISSUE THREAD

**Issue by AdminisTazzer** [Ylianst/MeshCentral#3095]
Hi,

I'm having an Issue with the Mesh Agent Database File on Windows endpoints which is located in the install directory, "C:\Program Files\Mesh\MeshAgent\MeshAgent.db". On some endpoints, especially noticeable on servers (it just happened on a Windows Server 2008 R2 Enterprise) the Data Base File tends to becomes really big. Last time I Reset it, it was up to 5Gb. Is there a way to prevent this, is this a bug that can be fixed ? 

If you need more Info about this, let me know.

Regards!

**Comment by Ylianst** [comment]
Are you using any plugins? I think there are plugin that will do that. Note that I don't test MeshCentral with any plugins.

**Comment by AdminisTazzer** [comment]
Yes we have a couple Plugins setup... do you, by any chance, know which ones are more likely to cause this?
These are the plugins We have installed and are all up to date.

![Screen Shot 2021-09-02 at 5 46 27 PM](https://user-images.githubusercontent.com/90004309/131920727-7e5e44ed-633b-4c85-870c-a8c97c28386a.png)

**Comment by krayon007** [comment]
If you pull the code from the agent repo, copy the file modules/dbTool.js into the folder where the agent is installed. Then run the command: 
`MeshAgent dbTool.js list`
This will list all the keys. We can start here to see if the db has a large number of keys.

**Comment by Ylianst** [comment]
There have been reports of this before. You can ping @ryanblenis who works on plug-ins, but this is not something I can help with.

**Comment by ryanblenis** [comment]
> Yes we have a couple Plugins setup... do you, by any chance, know which ones are more likely to cause this?
> These are the plugins We have installed and are all up to date.
> 
> ![Screen Shot 2021-09-02 at 5 46 27 PM](https://user-images.githubusercontent.com/90004309/131920727-7e5e44ed-633b-4c85-870c-a8c97c28386a.png)

It's EventLog. EventLog sends event logs from your endpoints to the server where they are stored for 30 days for access/retrieval. So you've got 30 days times the number of endpoints (and the number of entries per endpoint) for the logs that you're watching being stored in that database.

**Comment by AdminisTazzer** [comment]
ok, but I mean that the Db File that becomes really big is the one stored on the endpoints not on our Mesh Central Server.
(some of our endpoints are servers) The server itself seems fine, its the really endpoints that get bloated.

**Comment by Ylianst** [comment]
Yes, we are talking about the meshagent.db on the endpoint. The "EventLog" plugin like other plugins modified the code that the agent runs and can cause this behavior.

**Comment by AdminisTazzer** [comment]
All right, Thanks a lot I will look into that!

**Comment by ryanblenis** [comment]
> ok, but I mean that the Db File that becomes really big is the one stored on the endpoints not on our Mesh Central Server.
> (some of our endpoints are servers) The server itself seems fine, its the really endpoints that get bloated.

Ah, the endpoint, not the server. Are you on the latest version of EventLog? I was under the impression that issue for the endpoints was fixed. If you are, can you compress and send me the file so I can see what's being written that is causing the bloat?
